### PR TITLE
fix(candevice): add empty device struct for non-linux

### PR DESCRIPTION
**Pull request by saimanwong**
Fixes the following for non-linux host when running make:

```
[go-lint]
pkg/candevice/device_others.go:10:22: undeclared name: `Device` (typecheck)
func New(_ string) (*Device, error) {
^
[go-lint] exit status 1
[go-review] exit status 1
```
---

Awesome that other OSs are taken into consideration and the not supported error is returned. With this change, the error will now get propagated to the non-linux user when using the `candevice` package because the `Device` type will exist.

### CODE REVIEW

**Review by hashemmm96 — APPROVED** [review]
Nice find! I've created #110 to fix the failing CI.

**Comment by saimanwong** [comment]
> Nice find! I've created #110 to fix the failing CI.

Gotchya, no rush at all 👍  Will wait for #110 to gets merged before rebasing and merging this one.

**Comment by hashemmm96** [comment]
> > Nice find! I've created #110 to fix the failing CI.
> 
> Gotchya, no rush at all +1 Will wait for #110 to get merged before rebasing and merging this one.

Should be possible to rebase & merge now :)

**Pull request merged**
